java/core/src/java/org/apache/orc/OrcConf.java

```diff
@@ -72,6 +72,18 @@ public enum OrcConf {
       "Define the compression strategy to use while writing data.\n" +
           "This changes the compression level of higher level compression\n" +
           "codec (like ZLIB)."),
+  COMPRESSION_ZSTD_LEVEL("orc.compression.zstd.level",
+      "hive.exec.orc.compression.zstd.level", 3,
+      "Define the compression level to use with ZStandard codec "
+          + "while writing data."),
+  COMPRESSION_ZSTD_WINDOWLOG("orc.compression.zstd.windowlog",
+      "hive.exec.orc.compression.zstd.windowlog", 0,
+      "Set the maximum allowed back-reference distance for "
+          + "ZStandard codec, expressed as power of 2."),
+  COMPRESSION_ZSTD_LONGMODE("orc.compression.zstd.longmode",
```

Comment:
Have you tried this before? I'm not sure if it really works here since ORC usually does not employ large compression pages.

Reply:
source table: ORC zlib 4408374802439 4TB

orc.compression.zstd.windowlog=0
compress size: 3119313447131 2905G

orc.compression.zstd.windowlog=27
compress size: 3036659575064 2828G